File: src/main/java/com/example/RESTAPIarticle/entity/Magazine.java
package com.example.RESTAPIarticle.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;

import java.util.ArrayList;
import java.util.List;

import static javax.persistence.CascadeType.ALL;

@Entity
@Table(name="Magazine")
public class Magazine {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name="id")
    private int id;

    @Column(name="name")
    private String name;

    @OneToMany(mappedBy = "magazine", cascade = { ALL })
    @JsonIgnore
    private List<Article> articles;

    public Magazine() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public List<Article> getArticles() {
        return articles;
    }

    public void setArticles(List<Article> articles) {
        this.articles = articles;
    }

    public void add(Article tempArticle) {
        if(articles == null) {
            articles = new ArrayList<>();
        }

        articles.add(tempArticle);
        tempArticle.setMagazine(this);
    }

    @Override
    public String toString() {
        return "Magazine{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", articles=" + articles +
                '}';
    }
}

File: src/main/java/com/example/articles/entity/Magazine.kt
package com.example.articles.entity

import com.example.articles.controller.MagazineResponse
import org.hibernate.Hibernate
import java.util.Collections.emptyList
import javax.persistence.*
import kotlin.streams.toList

@Entity
@Table(name = "Magazine")
data class Magazine(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "Id")
    val id: Int,

    @Column(name = "Name")
    val name: String,

    @OneToMany(mappedBy = "magazine", cascade = [CascadeType.ALL])
    val articles: MutableList<Article>
) {
    constructor() : this(
        0,
        "asd",
        emptyList<Article>()
    )

    fun toResponse(): MagazineResponse {
        return MagazineResponse(
            this.id,
            this.name,
            this.articles.stream().map { it.toResponse() }.toList()
        )
    }


    fun add(tempArticle: Article) {
        articles.add(tempArticle)
        tempArticle.magazine = this
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as Magazine

        return id == other.id
    }

    override fun hashCode(): Int = hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , name = $name )"
    }
}